yt_dlp/extractor/linkedin.py

```diff
@@ -104,12 +104,17 @@ def _real_extract(self, url):
         like_count = int_or_none(get_element_by_class('social-counts-reactions__social-counts-numRections', webpage))
         creator = strip_or_none(clean_html(get_element_by_class('comment__actor-name', webpage)))
 
-        sources = self._parse_json(extract_attributes(self._search_regex(r'(<video[^>]+>)', webpage, 'video'))['data-sources'], video_id)
+        video_json = extract_attributes(self._search_regex(r'(<video[^>]+>)', webpage, 'video'))
+        sources = self._parse_json(video_json['data-sources'], video_id)
         formats = [{
             'url': source['src'],
             'ext': mimetype2ext(source.get('type')),
             'tbr': float_or_none(source.get('data-bitrate'), scale=1000),
         } for source in sources]
+        subtitles = {}
+        sub_url = video_json.get('data-captions-url')
+        if sub_url:
+            subtitles.setdefault('en', []).append({'url': sub_url, 'ext': 'vtt'})
```

Comment:
I'd do:
```suggestion
        subtitles = {'en': [{
            'url': video_attrs['data-captions-url'],
            'ext': 'vtt',
        }]} if url_or_none(video_attrs.get('data-captions-url')) else {}
```

Reply:
With @bashonly's suggestion, and for brevity, I would have reduced both your suggestions to:
```suggestion
        subtitles = {}
        if sub_url := url_or_none(video_attrs.get('data-captions-url')):
            subtitles = {'en': [{'url': sub_url, 'ext': 'vtt'}]}
```
This avoids resolving `video_attrs['data-captions-url']` twice, which is just a matter of style. Do you agree this is okay?
(`setdefault()` obviously isn't really needed.)

I can of course also break the `subtitles` assignment dict into several lines.
```suggestion
        subtitles = {}
        if sub_url := url_or_none(video_attrs.get('data-captions-url')):
            subtitles = {'en': [{
                'url': sub_url,
                'ext': 'vtt',
            }]}
```

Reply:
I think pukkandan's suggestion is the cleanest